run_matrix.py

```diff
@@ -174,30 +174,28 @@ def _get_fixture(charm_config: "_CharmTestConfig", charm_path: Path) -> FixtureS
 
 def _pre_run(charm_config: "_CharmTestConfig", charm_path: Path):
     """Run whatever setup commands were configured in the custom test config."""
-    if charm_config.test_setup:
-        error = None
-        if pre_run_cfg := charm_config.test_setup.get("pre_run"):
-            logging.info("Running custom pre_run script...")
-            try:
-                proc = Popen(
-                    pre_run_cfg,
-                    cwd=charm_path,
-                    shell=True,
-                    text=True,
-                    stderr=subprocess.PIPE,
-                )
-                proc.wait(timeout=60)  # wait for up to 60 seconds
-
-                if proc.returncode != 0:
-                    error = proc.stderr.read()
-            except Exception:
-                error = str(f"uncaught exception raised by subprocess: {error}")
-
-        if error:
+    if not charm_config.test_setup:
+        return
+    if pre_run_cfg := charm_config.test_setup.get("pre_run"):
+        logging.info("Running custom pre_run script...")
+        try:
+            proc = subprocess.run(
```

Comment:
since we only need the stdout, why not `getoutput`?

Reply:
I think we're not meant to use `getoutput` any more, but we could use `check_output`, yes. I guess it depends whether we want to distinguish between stdout/stderr - at the moment it's putting stderr in the exception and stdout in the debug log but just putting the combined output in both is probably just as useful in this case.